api/capi/include/nntrainer.h

```diff
@@ -35,6 +35,10 @@
 #ifdef __cplusplus
 extern "C" {
 #endif /* __cplusplus */
+
+typedef void *vector_of_tensordims_h; /// Not necessarily, but something like
```

Comment:
Below options are eligible. (please throw more)

## Option 1. using integer array + size

### Pros

1. straight forward

### Cons

1. not extensible as required

## Option 2. using `ml_tensors_info`, 

### Pros

1. using full fledged structure + thread safe public api will reduce implementation time 
1. extensible
1. minimize overlaps

### Cons

1. This api will require `capi-ml-inference`. This tensor / api is not defined at `capi-ml-common`

## Option 3. design a dedicated structure

### Pros
1. Extensible
1. Will be tailored to our requirements.

### Cons
1. Might going to overlap
1. Will take more time and error-prone.


I'd like to get opinions regarding this.

Reply:
I highly recommend option 2. You can discuss wtih @myungjoo to move ml_tensors_info to capi-ml-common, if you dont want dependency on capi-ml-inference.
IMO its not a good design for capi-ml-inference and capi-ml-training to have different tensors info data structures. I think We should keep compatibility between the two APIs as much as possible.

Reply:
We've discussed offline to go for option 2. There will be changes accordingly within this PR.